### PR TITLE
Add NetLog record method to Cobalt panel in Devtools

### PR DESCRIPTION
**Pull request by sideb0ard**
Much like the utility button to capture and download a Trace, this CL adds a button to the Cobalt tab of Devtools for capturing a NetLog.

View it here - https://screenshot.googleplex.com/ATA2L2GcEH6W272 

See here for more details on NetLog - https://github.com/youtube/cobalt/blob/main/cobalt/doc/net_log.md

b/305257093

### CODE REVIEW

File: cobalt/h5vcc/h5vcc_net_log.cc [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+namespace h5vcc {
+
+namespace {
+const char* kOutputNetLogFilename = "h5vcc_netlog.json";
+
+base::FilePath filepath_to_absolute(
+    const base::FilePath& output_path_relative_to_logs) {
+  base::FilePath result;
+  base::PathService::Get(cobalt::paths::DIR_COBALT_DEBUG_OUT, &result);
+  return result.Append(output_path_relative_to_logs);
```
Comment (jellefoks) [inline cobalt/h5vcc/h5vcc_net_log.cc:31]: Let's make sure this can't be abused to read any file off the filesystem.

Comment (sideb0ard) [reply cobalt/h5vcc/h5vcc_net_log.cc]: cools, yeah, no need to make this configurable, I've removed that option to pass a filename.

File: cobalt/h5vcc/h5vcc_net_log.cc [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+namespace h5vcc {
+
+H5vccNetLog::H5vccNetLog(cobalt::network::NetworkModule* network_module)
+    : network_module_{network_module} {}
+
+void H5vccNetLog::Start() { network_module_->StartNetLog(); }
+
+void H5vccNetLog::Stop() { network_module_->StopNetLog(); }
+
+std::string H5vccNetLog::Read() {
```
Comment (jellefoks) [inline cobalt/h5vcc/h5vcc_net_log.cc:32]: Since a required side effect is stopping the netlog, a more accurate name would be `StopAndRead()`.

Comment (sideb0ard) [reply cobalt/h5vcc/h5vcc_net_log.cc]: done!

**Review by jellefoks — APPROVED** [review]

File: third_party/devtools/front_end/cobalt/cobalt.js [hunk 3]
```diff
@@ -88,6 +115,7 @@ export default class CobaltPanel extends UI.VBox {
         lifecycleContainer.appendChild(UI.createTextButton(Common.UIString('Quit'), event => {
             this._cobaltAgent.invoke_sendConsoleCommand({ command: 'quit' });
         }));
+
```
Comment (kaidokert) [inline third_party/devtools/front_end/cobalt/cobalt.js:118]: nit: commit noise :)

File: third_party/devtools/front_end/cobalt/cobalt.js [hunk 4]
```diff
@@ -69,6 +95,7 @@ export default class CobaltPanel extends UI.VBox {
                 command: 'debug_log', message: 'off'
             });
         }));
+
```
Comment (kaidokert) [inline third_party/devtools/front_end/cobalt/cobalt.js:98]: commit noise:)

File: cobalt/h5vcc/h5vcc.idl [hunk 5]
```diff
@@ -41,6 +41,7 @@ interface H5vcc {
   readonly attribute H5vccStorage storage;
   readonly attribute H5vccSystem system;
   readonly attribute H5vccTraceEvent traceEvent;
+  readonly attribute H5vccNetLog netLog;
```
Comment (kaidokert) [inline cobalt/h5vcc/h5vcc.idl:44]: I've mentioned it offline, but do consider if this needs to be a standalone toplevel API - could just put it under traceEvent to reduce required boilerplate. 

It's probably cleaner this way though, fine to leave as is.

**Review by kaidokert — APPROVED** [review]

File: third_party/devtools/front_end/cobalt/cobalt.js (hunk 3 above)
Comment (sideb0ard) [reply third_party/devtools/front_end/cobalt/cobalt.js:118]: it's spacious, cleaner to read! :)

**Pull request merged**